### PR TITLE
Better handling of template urls and config.xml locations

### PR DESCRIPTION
**Pull request by chrisgriffith**
Will now copy config.xml files outside the www folder

Handles alternate template URLS.

### CODE REVIEW

File: lib/create.js [hunk 1]
```diff
@@ -200,10 +200,12 @@ module.exports.deleteInvalidTemplate = function(options) {
...
-module.exports.downloadTemplate = function(options, callback) {
-    var destinationPath = module.exports.templatePath(options),
-        url = 'https://github.com/phonegap/phonegap-app-hello-world/archive/master.tar.gz';
-
+module.exports.downloadTemplate = function(options, callback, url) {
+    var destinationPath = module.exports.templatePath(options);
+	if (url === undefined) {
+        var url = 'https://github.com/phonegap/phonegap-app-hello-world/archive/master.tar.gz';
+    }
+
```
Comment (mwbrooks) [inline lib/create.js]: What's the purpose of this new addition? As far as I can tell, `url` will always be undefined.

Also, the function pattern should be `(options, callback)` where callback is always the last argument. If `url` is needed, then it should be `options.url`.

File: lib/create.js [hunk 2]
```diff
@@ -281,6 +283,12 @@ module.exports.createProject = function(options, callback) {
     };
     if (fs.existsSync(configXML.wwwPath)) {
         fs.renameSync(configXML.wwwPath, configXML.projectPath);
+    } else {
+        configXML = {
+            projectPath: path.join(options.path, 'config.xml'),
+            sourcePath: path.join(module.exports.templatePath(options), 'config.xml')
+        };
+        fs.renameSync(configXML.sourcePath, configXML.projectPath);
```
Comment (mwbrooks) [inline lib/create.js]: Nice

**Comment by mwbrooks** [comment]
Thanks @chrisgriffith,

I left some feedback in the commit messages.

Also, we ned to have test passing and new tests added to cover the config.xml scenario.

File: lib/create.js (hunk 1 above)
Comment (chrisgriffith) [inline lib/create.js]: That seems fine. I know the PhoneGap Desktop App is going to support multiple templates in the near future and was addressing that issue early.

**Comment by chrisgriffith** [comment]
My testing mojo is low, so I will have to leave that on your shoulders.

File: lib/create.js (hunk 1 above)
Comment (mwbrooks) [inline lib/create.js]: Ah alright. I think we can remove this chunk of code entirely. I'm a big fan of discrete and focused work and the multiple templates is out of scope of patching the config.xml. Also, the PG Desktop App will likely be using the PG CLI for templates and the `connect-phonegap` create will be deprecated entirely. So we may never need to add template support to this function.

**Comment by mwbrooks** [comment]
haha, alright. I've got a backlog of work, but I'll queue this sucker up! Thanks for kicking it off.

**Comment by chrisgriffith** [comment]
NP. It was the least I could do after killing the Desktop app by accident ;)

Haven't looked at the &gt; issue in the generated config.xml yet. Figured I
would let the experts handle that one

cg

**Comment by filmaj** [comment]
Looks like there are conflicts here. Is this still valid?

**Comment by purplecabbage** [comment]
It this still valued? If so, please rebase and reopen.

**Pull request closed without merging**